components/contacts/contact-form.js

```diff
@@ -0,0 +1,162 @@
+/* global fetch */
+
+import 'whatwg-fetch';
+import React from 'react';
+import PropTypes from 'prop-types';
+
+class ContactForm extends React.Component {
+  state = ({
+    firstname: {
+      value: '',
+      error: false,
+    },
+    lastname: {
+      value: '',
+      error: false,
+    },
+    email: {
+      value: '',
+      error: false,
+    },
+    phone: {
+      value: '',
+      error: false,
+    },
+    message: {
+      value: '',
+      error: false,
+    },
+  })
+  onSubmit = (e) => {
+    e.preventDefault();
+    const { callback } = this.props;
+
+    callback({ isPending: true });
+
+    fetch('/contact', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(this.state),
+    }).then(response => response.json()).then((json) => {
```

Comment:
you need to handle error [read this](https://www.npmjs.com/package/whatwg-fetch#handling-http-error-statuses)